framework/src/main/java/org/checkerframework/common/aliasing/AliasingVisitor.java

```diff
@@ -263,15 +264,39 @@ protected void checkThisOrSuperConstructorCall(
 
     /**
      * Returns true if {@code exp} has type {@code @Unique} and is not a method invocation nor a new
-     * class expression.
+     * class expression. It checks whether the tree expression is unique by either checking for an
+     * explicit annotation or checking whether the class of the tree expression {@code exp} has type
+     * {@code @Unique}
      *
      * @param exp the Tree to check
      */
     private boolean canBeLeaked(Tree exp) {
         AnnotatedTypeMirror type = atypeFactory.getAnnotatedType(exp);
         boolean isMethodInvocation = exp.getKind() == Kind.METHOD_INVOCATION;
         boolean isNewClass = exp.getKind() == Kind.NEW_CLASS;
-        return type.hasExplicitAnnotation(Unique.class) && !isMethodInvocation && !isNewClass;
+        boolean isUniqueType = (isUniqueClass(type) || type.hasExplicitAnnotation(Unique.class));
```

Comment:
The outer parenthesis aren't necessary.

Reply:
I have removed the parenthesis in my latest commit.